src/providers/ObservationProvider.js

```diff
@@ -34,19 +31,19 @@ const ObservationProvider = ( { children }: Props ): Node => {
         setObservationList( localObservations );
       }
 
-      // Live queries and objects emit notifications when something has changed that we can listen for.
-      subscriptionRef.current = localObservations;
+      localObservations.addListener( ( ) => {
+        // If you just pass localObservations you end up assigning a Results
+        // object to state instead of an array of observations. There's
+        // probably a better way...
+        setObservationList( localObservations.map( o => o ) );
```

Comment:
Hah, I was hoping you could tell me. In practical terms, if I did *not* do this, `observationsList` would get assigned a [Results object](https://docs.mongodb.com/realm-legacy/docs/javascript/latest/api/Realm.Results.html), not an array. In the test environment, that yielded a warning about not wrapping updates to state in `act()`, and indeed, it didn't actually update the state in a way that the test would wait for. In the app itself, everything was fine. You can try it by changing this line and running `MyObservations.test.js`. I'd prefer not to mock parts of Realm since this is an integration test, but it seemed like Realm was performing some asynchronous task that the test wasn't able to wait for, so this was my compromise.

Reply:
Hmm. I think this is fine for now, but it's something we might keep an eye on to make sure there are no performance implications.

It's also a little strange to me that we're setting state with the Results object on line 31, but resetting state with an array on line 38. Probably better to be consistent?

Reply:
> Probably better to be consistent?

Not sure. Personally, I think the discrepancy and comment point out that something is wrong here, so it's better to isolate that wrongness in a single place where we can annotate it and hopefully fix it some day.